src/resolve/resolved-link-bus.c

```diff
@@ -640,6 +640,7 @@ int bus_link_method_set_dns_over_tls(sd_bus_message *message, void *userdata, sd
 
         if (l->dns_over_tls_mode != mode) {
                 link_set_dns_over_tls_mode(l, mode);
+                link_allocate_scopes(l);
```

Comment:
Why not call this in `link_set_dns_over_tls_mode()`?

Reply:
`link_set_dns_over_tls_mode()` is called on sd-network change, and in that case, `link_allocate_scope()` will be called after all sd-network properties are parsed.